[CUBRID-24687] query result is strange, when there is hidden columns in the select list.

mhoh3963 opened the pull request:
http://jira.cubrid.org/browse/CBRD-24687

**Description**
The query result is strange, when there is hidden columns in ther select list. This is the case when sp is used in the order by clause, as in the following query. (ex : SELECT col1, col2) FROM tbl1 ORDER BY intTest(col1))

**Resolution**
- fix to use query_type instead of col_cnt at db_query_column_count().

mhoh3963 committed:
[CUBRID-24687] query result is strange, when there is hidden columns in the select list

http://jira.cubrid.org/browse/CBRD-24687

**Description**
The query result is strange, when there is hidden columns in ther select list.
This is the case when sp is used in the order by clause, as in the following query.
(ex : SELECT col1, col2) FROM tbl1 ORDER BY intTest(col1))

**Resolution**
- fix to replace q_result->num_column instead of db_query_column_count() at cur_tuple().
- fix to use query_type instead of col_cnt at db_query_column_count().

beyondykk9 commented on src/compat/db_query.c:
```diff
@@ -3164,7 +3167,12 @@ db_query_column_count (DB_QUERY_RESULT * result)
     }
 
-  return (DB_OID_INCLUDED (result)) ? (result->col_cnt - 1) : result->col_cnt;
+  for (t = result->query_type; t != NULL; t = db_query_format_next (t), num_attrs++)
```
```suggestion
  for (t = result->query_type; t != NULL; t = db_query_format_next (t))
     {
       num_attrs++;
     }
```

mhoh3963 replied:
Thank you, I accept.

beyondykk9 commented on src/compat/db_query.c:
```diff
@@ -3164,7 +3167,12 @@ db_query_column_count (DB_QUERY_RESULT * result)
+      ;
+    }
+
+  return (num_attrs);
```
```suggestion
  return num_attrs;
```

mhoh3963 replied:
Thank you, I accept.

beyondykk9 commented on src/compat/db_query.c:
```diff
@@ -3150,6 +3150,9 @@ db_query_tuple_count (DB_QUERY_RESULT * result)
 db_query_column_count (DB_QUERY_RESULT * result)
 {
+  DB_QUERY_TYPE *t;
+  int num_attrs = 0;
```
the variable "num_cols" would be better than "num_attrs"

mhoh3963 replied:
Thank you, I accept.

beyondykk9 commented on src/broker/cas_execute.c:
```diff
@@ -4914,7 +4914,7 @@ cur_tuple (T_QUERY_RESULT * q_result, int max_col_size, char sensitive_flag, DB_
   int err_code;
 
-  ncols = db_query_column_count (result);
+  ncols = q_result->num_column;
```
it would be better add a comment to explain the logic

mhoh3963 replied:
This code was occurred a problem when executing the call and evaluation statement. So I will revert it.

mhoh3963 committed:
- apply reviewer's opinion
- Fix the column count could not be gotten, when executing call and evaluate statements.

mhoh3963 committed:
- remove bracket
